src/jemalloc.c

```diff
@@ -2105,6 +2105,9 @@ imalloc_no_sample(static_opts_t *sopts, dynamic_opts_t *dopts, tsd_t *tsd,
 		arena = NULL;
 	} else {
 		arena = arena_get(tsd_tsdn(tsd), dopts->arena_ind, true);
+		if (unlikely(arena == NULL)) {
```

Comment:
I'm kinda neutral on this. There are two cases:
- The arena ID is a user-created arena, in which case it's a usage error if the user didn't already initialize it, and we are free to assert-fail, misbehave, etc.
- The arena ID is an auto-arena that simply hasn't been initialized yet. We already reserve the right to re-route auto arena requests (like for the huge arena or for reentrant allocations), and will probably stop listening to them entirely at some point.

No real preference either way though.

Reply:
Makes sense. I spotted this because we're not handling it consistently: in `malloc()` we fall back but in `realloc()` we OOM. I think we at least need to make them consistent.

I'm changing it so that (by checking the id against `narenas_auto`) we fail if it's a manually created arena and we fallback if it's an auto arena. Does that sound better?

Reply:
TBH I think anything is fine; I was just giving a little context on the history. Maybe `assert(arena != NULL || arena_ind < opt_narenas)` and whatever's easiest after that. Surfacing the OOM earlier than strictly necessary is fine IMO.

Reply:
Please let me know what you think of my new revision. BTW I think it'd be `narenas_auto` rather than `opt_narenas` to be more rigorous, though the two should rarely be different.